From 8e7d73797eb9a7e3b1044d27a343f3c21387b8b4 Mon Sep 17 00:00:00 2001
From: Til Kaiser <mail@tk154.de>
Date: Tue, 29 Oct 2024 09:21:53 +0100
Subject: [PATCH] util/xdp_sample: remove <sys/sysinfo.h> for musl
 compatibility

In the musl libc environment, `<sys/sysinfo.h>` also defines
the `sysinfo` struct, causing a redefinition conflict:

toolchain-x86_64_gcc-14.2.0_musl/include/linux/kernel.h:5,
                 from ../../headers/linux/hashtable.h:15,
                 from xdp_sample.c:33:
toolchain-x86_64_gcc-14.2.0_musl/include/linux/sysinfo.h:8:8: error: redefinition of 'struct sysinfo'
    8 | struct sysinfo {
      |        ^~~~~~~
In file included from xdp_sample.c:24:
toolchain-x86_64_gcc-14.2.0_musl/include/sys/sysinfo.h:10:8: note: originally defined here
   10 | struct sysinfo {
      |        ^~~~~~~

This commit removes the `<sys/sysinfo.h>` include. It seems that
none of its symbols are used in the xdp_sample code anyway.

Signed-off-by: Til Kaiser <mail@tk154.de>
---
 lib/util/xdp_sample.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/util/xdp_sample.c b/lib/util/xdp_sample.c
index 33c9f58..77c74d6 100644
--- a/lib/util/xdp_sample.c
+++ b/lib/util/xdp_sample.c
@@ -21,7 +21,6 @@
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
 #include <bpf/libbpf.h>
-#include <sys/sysinfo.h>
 #include <sys/timerfd.h>
 #include <sys/utsname.h>
 #include <linux/limits.h>
-- 
2.47.0

